.pre-commit-config.yaml

```
@@ -1,33 +1,13 @@
 repos:
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v3.15.0
-    hooks:
-      - id: pyupgrade
-        args: [--py38-plus]
-
-  - repo: https://github.com/psf/black-pre-commit-mirror
-    rev: 23.12.1
-    hooks:
-      - id: black
-        args: [--skip-string-normalization]
-
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.13.2
-    hooks:
-      - id: isort
-        args: [--profile=black]
-
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.1.0
-    hooks:
-      - id: flake8
-        additional_dependencies:
-          [flake8-2020, flake8-implicit-str-concat]
-
-  - repo: https://github.com/pre-commit/pygrep-hooks
-    rev: v1.10.0
-    hooks:
-      - id: python-check-blanket-noqa
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.5.7
+    hooks:
+      - id: ruff-format
+        name: Run Ruff (format)
+        args: [--check]
+      - id: ruff
+        name: Run Ruff (lint)
+        args: [--exit-non-zero-on-fix]
```

Ruff's pre-commit docs say:

> When running with `--fix`, Ruff's lint hook should be placed _before_ Ruff's formatter hook, and
_before_ Black, isort, and other formatting tools, as Ruff's fix behavior can output code changes
that require reformatting.

https://github.com/astral-sh/ruff-pre-commit#using-ruff-with-pre-commit

```suggestion
      - id: ruff
        name: Run Ruff (lint)
        args: [--exit-non-zero-on-fix]
      - id: ruff-format
        name: Run Ruff (format)
        args: [--check]
```